torchvision/ops/dice_loss.py

```diff
@@ -0,0 +1,69 @@
+import torch
+import torch.nn.functional as F
+
+from ..utils import _log_api_usage_once
+
+
+def dice_loss(inputs: torch.Tensor, targets: torch.Tensor, reduction: str = "none", eps: float = 1e-8) -> torch.Tensor:
+    r"""Criterion that computes Sørensen-Dice Coefficient loss.
+
+    We compute the Sørensen-Dice Coefficient as follows:
+
+    .. math::
+
+        \text{Dice}(x, class) = \frac{2 |X \cap Y|}{|X| + |Y|}
+
+    Where:
+       - :math:`X` expects to be the scores of each class.
+       - :math:`Y` expects to be thess tensor with the class labels.
+
+    the loss, is finally computed as:
+
+    .. math::
+
+        \text{loss}(x, class) = 1 - \text{Dice}(x, class)
+
+    Args:
+        inputs: (Tensor): A float tensor of arbitrary shape.
+                The predictions for each example.
+        targets: (Tensor): A float tensor with the same shape as inputs. Stores the binary
+                classification label for each element in inputs
```

Comment:
Tbh this is a one hot tensor. We expect users to pass one hot tensor which is completely fine. I would add the word one hot to emphasis this.

Reply:
Hmm, I changed it to binary from one hot, because I believe one hot would imply we support multi-class loss too.

Reply:
Look at my comment below,  one hot is a binary tensor. It just a convenient name :smile: as it implies user can use F.one_hot.
Although there are other ways to create one hot tensor too. It's upto the user. He could even set a manual threshold and do one hotting.